src/main.rs

```diff
@@ -1,3 +1,104 @@
+#[macro_use]
+extern crate log;
+
+use std::collections::HashMap;
+use std::io::{BufRead, BufReader, BufWriter, Write};
+use std::net::{Shutdown, TcpListener, TcpStream};
+use std::sync::{Arc, RwLock};
+use std::{thread, time};
+
+use crate::message::{Assignment, Get};
+
+pub mod message;
+pub mod types;
+
+const PORT: u16 = 4333;
+
+fn handle_client(stream: TcpStream, _counter: Arc<RwLock<HashMap<String, String>>>) {
+    let mut reader = BufReader::new(&stream);
+    let mut writer = BufWriter::new(&stream);
+    let mut buffer = Vec::new();
+    'read: while match reader.read_until(b'\n', &mut buffer) {
+        Ok(size) => {
+            if size == 0 {
+                break 'read;
+            }
+            trace!("stream read {} bytes", size);
+
+            let _get = match Get::deserialize(&buffer[..size]) {
+                Ok(message) => message,
+                Err(e) => {
+                    error!("deserialization failed: {}", e);
+                    continue 'read;
+                }
+            };
+
+            let assignment = Assignment {
+                addresses: vec![
```

Comment:
Consider making this list of task servers a global constant, it will likely be needed in other places.

Reply:
Tabling this for now, arrays of strings cause issues.